### PR TITLE
feat: narakeet integration

### PR DESCRIPTION
**Pull request by MySmilingTurban**
The backend server which it calls is currently hosted on Heroku as an Express API and its code can be seen in [backend/index.js](https://github.com/KhalisFoundation/gurmukhi-web/blob/75-task-setup-narakeet-for-text-to-speech/backend/index.js)
This PR consists code to integrate our web app with Narakeet TTS API, below is a little demo for the same

https://github.com/KhalisFoundation/gurmukhi-web/assets/51584037/8d530126-f6d3-4d13-9d01-c1d62a0e4f45

UPDATE:
The express API backend which connects our web app to Narakeet is now deployed on Firebase Functions and is working as intended.

One of the issues which I am facing currently is updating the audio for question in front-end, as whenever we go to next question the options audio update but question one's remain same - we'll be good to go once this is resolved.

### CODE REVIEW

**Comment by amitojsingh** [comment]
@MySmilingTurban Veer ji can we host this on firebase?

File: src/utils/textToSpeech.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// Import other required libraries
+import fs from 'fs';
+import util from 'util';
+// Creates a client
+const client = new TextToSpeechClient();
+
+export async function getAudioContent(text: string) {
+  // Construct the request
+  const request: protos.google.cloud.texttospeech.v1.ISynthesizeSpeechRequest = {
```
Comment (amitojsingh) [inline src/utils/textToSpeech.ts:12]: I think we don't need google text to speech we can remove this file.

File: tsconfig.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "strict": true,
     "forceConsistentCasingInFileNames": true,
     "noFallthroughCasesInSwitch": true,
+    "noImplicitAny": false,
```
Comment (amitojsingh) [inline tsconfig.json:17]: why we made implicitAny false?

**Review by amitojsingh — CHANGES_REQUESTED** [review]

File: src/components/buttons/TextToSpeechBtn.tsx [hunk 3]
```diff
@@ -1,15 +1,76 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { FC, useEffect, useRef, useState } from 'react';
+import { generateNarakeetAudio } from 'narakeet-axios';
 
 interface TextToSpeechBtnProps {
+  text?: string;
```
[the remark this answers is not in the document]
Comment (MySmilingTurban) [reply src/components/buttons/TextToSpeechBtn.tsx:5]: hnji, fixed

File: src/components/buttons/TextToSpeechBtn.tsx [hunk 4]
```diff
@@ -1,15 +1,76 @@
...
+  word_id?: string;
   backgroundColor?: string;
 }
 
-export default function TextToSpeechBtn({ backgroundColor }: TextToSpeechBtnProps) {
+const TextToSpeechBtn: FC<TextToSpeechBtnProps> = ({ text = 'word', backgroundColor }) => {
+  const [isLoading, setIsLoading] = useState(false);
```
[the remark this answers is not in the document]
Comment (MySmilingTurban) [reply src/components/buttons/TextToSpeechBtn.tsx:11]: fixed

File: src/components/questions/multiple-choice.tsx [hunk 5]
```diff
@@ -116,7 +115,7 @@ export default function MultipleChoiceQuestion({
         <h1 className={'text-5xl gurmukhi text-black font-semibold'}>
           {highlightWord(question.question, question.word)}
         </h1>
-        <TextToSpeechBtn backgroundColor='bg-white-175' />
+        <TextToSpeechBtn backgroundColor='bg-white-175' text={question.question} />
```
[the remark this answers is not in the document]
Comment (MySmilingTurban) [reply src/components/questions/multiple-choice.tsx:118]: sure, fixed

File: src/utils/textToSpeech.ts (hunk 1 above)
Comment (MySmilingTurban) [reply src/utils/textToSpeech.ts:12]: done

**Comment by amitojsingh** [comment]
@MySmilingTurban veer ji can we host this to firebase?

**Comment by MySmilingTurban** [comment]
> @MySmilingTurban veer ji can we host this to firebase?

Hnji, I am working on it
Looking into firebase functions for this

**Comment by MySmilingTurban** [comment]
@amitojsingh veerji, should we merge this?
as the code will remain the same, we'll only have to change the env var for backend API URL

**Review by amitojsingh — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -8,6 +8,7 @@
     "@fortawesome/free-regular-svg-icons": "^6.4.2",
     "@fortawesome/free-solid-svg-icons": "^6.4.2",
     "@fortawesome/react-fontawesome": "^0.2.0",
+    "@google-cloud/text-to-speech": "^5.0.2",
```
Comment (amitojsingh) [inline package.json:11]: I think we can remove this.

File: src/components/buttons/TextToSpeechBtn.tsx [hunk 7]
```diff
@@ -1,15 +1,115 @@
...
+        }
+        if (!audioUrl || audioNotWorking) {
+          setLoading?.(true);
+          setIsLoading(true);
+          const audioContent = await generateNarakeetAudio(betterText, type, setAudioUrl, id ?? undefined);
+          console.log('Narakeet Audio Content: ', audioContent);
+          // justSetAudioUrl = true;
+        }
+      } catch (error) {
+        console.error('Error generating audio:', error);
```
Comment (amitojsingh) [inline src/components/buttons/TextToSpeechBtn.tsx:45]: we need to implement Firebase crashlytics to view any error. console is not a good practice. we can create a separate PR/ Task for this.

File: src/components/buttons/TextToSpeechBtn.tsx [hunk 8]
```diff
@@ -1,15 +1,115 @@
...
+    }
+  }, [audioRef.current]);
+
   return (
-    <Link to='' className={ttsClassname}>
-      <img src={'/icons/speaker.svg'} alt='Text to Speech' width={24} height={24} />
-    </Link>
+    <button className={ttsClassname} onClick={onBtnClick} disabled={isLoading}>
+      {isLoading ? (
+        <svg
```
Comment (amitojsingh) [inline src/components/buttons/TextToSpeechBtn.tsx:87]: can we use loaderComponent here?

**Review by amitojsingh — CHANGES_REQUESTED** [review]

File: src/components/buttons/TextToSpeechBtn.tsx (hunk 7 above)
Comment (MySmilingTurban) [reply src/components/buttons/TextToSpeechBtn.tsx:45]: created new issue for this

File: package.json (hunk 6 above)
Comment (MySmilingTurban) [reply package.json:11]: fixed

File: src/components/buttons/TextToSpeechBtn.tsx (hunk 8 above)
Comment (MySmilingTurban) [reply src/components/buttons/TextToSpeechBtn.tsx:87]: sure, fixed

**Review by amitojsingh — APPROVED** [review]

**Pull request merged**